Gems/Atom/RPI/Code/Source/RPI.Reflect/Image/StreamingImageAssetHandler.cpp

```diff
@@ -15,6 +15,106 @@ namespace AZ
 {
     namespace RPI
     {
+        StreamingImageAssetReloader::StreamingImageAssetReloader(Data::AssetId assetId, CompleteCallback callback)
+        {
+            m_imageAssetId = assetId;
+            m_completeCallback = callback;
+
+            // Get current image asset
+            m_imageAsset = AZ::Data::AssetManager::Instance().GetAsset<StreamingImageAsset>(assetId, AZ::Data::AssetLoadBehavior::Default);
+
+            // trigger reload for all its mip chains
+            if (m_imageAsset->GetMipChainCount() > 0)
+            {
+                for (uint32_t mipChainIndex = 0; mipChainIndex < m_imageAsset->GetMipChainCount() - 1; mipChainIndex++)
+                {
+                    auto& mipChainAsset = m_imageAsset->m_mipChains[mipChainIndex].m_asset;
+                    if (mipChainAsset.Get())
+                    {
+                        if (mipChainAsset.Reload())
+                        {
+                            m_pendingMipChainAssets.insert(mipChainAsset.GetId());
+                            Data::AssetBus::MultiHandler::BusConnect(mipChainAsset.GetId());
+                        }
```

Comment:
There are a lot of BusConnect / BusDisconnect calls happening in conditionals and specific locations. Could this all be simplified by just connecting unconditionally to all of them in the constructor, and disconnecting uncondtionally in the destructor? I'm not sure what this is trying to achieve in micro-managing the connections.